src/tests/BuildWasmApps/Wasm.Build.Tests/data/RunScriptTemplate.sh

```diff
@@ -4,13 +4,13 @@ EXECUTION_DIR=$(dirname $0)
 
 cd $EXECUTION_DIR
 
-if [ -z "$HELIX_WORKITEM_UPLOAD_ROOT" ]; then
+if [[ -z "$HELIX_WORKITEM_UPLOAD_ROOT" ]]; then
 	XHARNESS_OUT="$EXECUTION_DIR/xharness-output"
 else
 	XHARNESS_OUT="$HELIX_WORKITEM_UPLOAD_ROOT/xharness-output"
 fi
 
-if [ ! -z "$XHARNESS_CLI_PATH" ]; then
+if [[ -n "$XHARNESS_CLI_PATH" ]]; then
```

Comment:
Any reason to use [[ when most places changes use [

Reply:
It was discussed once with @janvorli and the consensus was to prefer `[[` over `[` in bash context: https://github.com/dotnet/runtime/pull/1753#discussion_r368626175.

I haven't changed all instances of `[[` -> `[` in bash scripts exhaustively in this PR, but mainly focused on those where we were mixing bash and unix shell syntax in bash scripts: `[[ $x = $y ]]` and `[ $x == $y ]`.